src/wp-includes/class-wp-theme.php

```diff
@@ -1993,16 +2001,38 @@ private function set_pattern_cache( array $patterns ) {
 			'version'  => $this->get( 'Version' ),
 			'patterns' => $patterns,
 		);
-		wp_cache_set( 'wp_theme_patterns_' . $this->stylesheet, $pattern_data, 'theme_files' );
+
+		/**
+		 * Filters whether to cache theme files persistently.
+		 *
+		 * @since 6.6.0
+		 *
+		 * @param bool   $cache_expiration Expiration time for the cache. Setting a value to `false` would bypass caching. Default `WP_Theme::$cache_expiration`.
+		 * @param string $context          Additional context for better cache control.
+		 */
+		$cache_expiration = apply_filters( 'wp_cache_theme_files_persistently', self::$cache_expiration, 'theme_block_patterns' );
```

Comment:
Why a filter for this? I realize there is `wp_cache_themes_persistently` in WP core, but if we don't reuse that same filter here, I don't see the benefit of a new filter. Where was introducing this filter discussed?

One of the reason to introduce the `theme_files` cache group instead of using the existing `themes` cache group is that the latter is subject to such a filter and therefore not persistent in most WordPress sites. If we now introduce a similar filter here, this decision would have been pointless.

As far as I recall, from what we previously discussed, anything in the `theme_files` group should _always_ be persistently cached. We were talking about making the expiration filterable, but not _whether_ to persistently cache it.

Reply:
@joemcgill I would be in favor of limiting the filter to purely controlling the TTL. Since it's a separate cache group that already behaves differently from the `themes` group, I don't think consistency with the other filter is important here. I don't see a good reason why we would want to allow opting out of persistent caching for this group. There is no such filter for any other WP core cache groups either.

Reply:
In that case, maybe even adding a filter for controlling the TTL of this group is unnecessary. Currently, the only way to filter the TTL for the `cache_expiration` property of the `WP_Theme` class is by passing an int value to the `wp_cache_themes_persistently` filter, so unless we want more granular control over the `theme_files` TTL than other theme related caches, the additional filter is probably overkill.

Reply:
@joemcgill @felixarntz  I have updated filter to `wp_cache_themes_persistently` and only using it for updating TTL.

Reply:
As agreed we have gone ahead and only implemented TTL-based control to manage cache_expiry for now.